sktime/datasets/_single_problem_loaders.py

```diff
@@ -1438,3 +1439,233 @@ def load_forecastingdata(
     return load_tsf_to_dataframe(
         path_to_file, replace_missing_vals, value_column_name, return_type
     )
+
+
+def load_m5(
+    extract_path=None,
+    folder_path=None,
+    include_events=False,
+    test=False,
+):
+    """Fetch M5 dataset from https://zenodo.org/records/12636070 .
+
+    Downloads and extracts dataset if not already downloaded. Fetched dataset is
+    in the standard .csv format and loaded into an sktime-compatible in-memory
+    format (`pd_multiindex_hier`).
+
+    Parameters
+    ----------
+    extract_path : str, optional (default=None)
+        The path to extract the data to. If the provided path doesn't
+        contain the data, the function creates a folder `m5-forecasting-accuracy`
+        in the provided path. If no path is provided, the function creates
+        the `m5-forecasting-accuracy` folder in the current working directory.
+        If a path is given, it can be absolute (e.g., C:/Temp) or
+        relative (e.g., Temp or ./Temp).
+
+    include_events : bool, optional (default=False)
+        If `True`, the resulting dataset will include additional columns
+        related to events. Including these columns allows for a richer
+        dataset that can be used to analyze the impact of events on sales.
+        If `False`, the dataset will exclude these columns, providing a
+        more streamlined version of the data.
+
+    test : bool, optional (default=False)
+        Loads a smaller part of the dataset which doesn't include events
+        for testing purposes. This should not be used in standard usage
+        but might be useful for developers running tests.
+
+    Returns
+    -------
+    data : pd.DataFrame of sktime type `pd_multiindex_hier`
+        The preprocessed dataframe containing the time series.
+
+    Dataset Description
+    --------------------
+    - **Number of Rows**: Approximately 58 million rows (for the full dataset).
+    - **Number of Columns**: Varies based on `include_events` parameter.
+      - Without events: 9 columns.
+      - With events: 13 columns.
+
+    Notes
+    -----
+    The dataset consists of three main files:
```

Comment:
something does not seem right. Should these not be loaded into three data frames?